datajunction-query/djqs/config.py

```diff
@@ -3,94 +3,202 @@
-from typing import Optional
+from enum import Enum
+from typing import Dict, List, Optional
 
 import toml
 import yaml
 from cachelib.base import BaseCache
 from cachelib.file import FileSystemCache
-from pydantic import BaseSettings
-from sqlmodel import Session, delete, select
 
-from djqs.exceptions import DJException
-from djqs.models.catalog import Catalog, CatalogEngines
-from djqs.models.engine import Engine
+from djqs.exceptions import DJUnknownCatalog, DJUnknownEngine
 
+_logger = logging.getLogger(__name__)
 
-class Settings(BaseSettings):  # pylint: disable=too-few-public-methods
+
+class EngineType(Enum):
     """
-    Configuration for the query service
+    Supported engine types
     """
 
-    name: str = "DJQS"
-    description: str = "A DataJunction Query Service"
-    url: str = "http://localhost:8001/"
+    DUCKDB = "duckdb"
+    SQLALCHEMY = "sqlalchemy"
+    SNOWFLAKE = "snowflake"
+    TRINO = "trino"
 
-    # SQLAlchemy URI for the metadata database.
-    index: str = "sqlite:///djqs.db?check_same_thread=False"
 
-    # The default engine to use for reflection
-    default_reflection_engine: str = "default"
+class EngineInfo:  # pylint: disable=too-few-public-methods
+    """
+    Information about a query engine
+    """
 
-    # The default engine version to use for reflection
-    default_reflection_engine_version: str = ""
+    def __init__(  # pylint: disable=too-many-arguments
+        self,
+        name: str,
+        version: str,
+        type: str,  # pylint: disable=redefined-builtin
+        uri: str,
+        extra_params: Optional[Dict[str, str]] = None,
+    ):
+        self.name = name
+        self.version = str(version)
+        self.type = EngineType(type)
+        self.uri = uri
+        self.extra_params = extra_params or {}
+
+
+class CatalogInfo:  # pylint: disable=too-few-public-methods
+    """
+    Information about a catalog
+    """
 
-    # Where to store the results from queries.
-    results_backend: BaseCache = FileSystemCache("/tmp/djqs", default_timeout=0)
+    def __init__(self, name: str, engines: List[str]):
+        self.name = name
+        self.engines = engines
 
-    paginating_timeout: timedelta = timedelta(minutes=5)
 
-    # How long to wait when pinging databases to find out the fastest online database.
-    do_ping_timeout: timedelta = timedelta(seconds=5)
+class Settings:  # pylint: disable=too-many-instance-attributes
+    """
+    Configuration for the query service
+    """
 
-    # Configuration file for catalogs and engines
-    configuration_file: Optional[str] = None
+    def __init__(  # pylint: disable=too-many-arguments,too-many-locals,dangerous-default-value
+        self,
+        name: Optional[str] = "DJQS",
+        description: Optional[str] = "A DataJunction Query Service",
+        url: Optional[str] = "http://localhost:8001/",
+        index: Optional[str] = "postgresql://dj:dj@postgres_metadata:5432/dj",
+        alembic_uri: Optional[
```

Comment:
Curious as to why there's a separation between alembic_uri and index?

Reply:
Thanks for this, I completely forgot that this can be removed now. It was initially different because alembic uses a sqlalchemy uri where the index value now gets passed directly into psycopg. The format is a little different where slqalchemy needs the `....+psycopg...` included.

Just updated this to removed `alembic_uri` since that's now set via a `--uri=....` parameter to the alembic command. This also helps with @sadath-12's requirement for multi-tenancy so he doesn't have to hack the Settings object. Another nice side-effect is that the alembic script now has no dependencies on the djqs library.